### PR TITLE
support SV82x platform

### PR DESCRIPTION
**Pull request by EndlessVoid**
Signed-off-by: zhouhuo <kael.chou@eeasytech.com>

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by EndlessVoid** [comment]
support  SV82x platform from Eeasytech

File: 3rdparty/SV82x/common/sample_comm_ippu.c [hunk 1]
```diff
@@ -0,0 +1,215 @@
+/*
+ *------------------------------------------------------------------------------
+ * @File      :    sample_comm_ippu.c
+ * @Date      :    2021-3-16
+ * @Author    :    lomboswer <lomboswer@lombotech.com>
+ * @Brief     :    Source file for MDP(Media Development Platform).
+ *
+ * Copyright (C) 2020-2021, LomboTech Co.Ltd. All rights reserved.
```
Comment (codingspirit) [inline 3rdparty/SV82x/common/sample_comm_ippu.c:8]: What's the license of those files? Are those files contributed by Eeasy?

File: source/SV82x/SV82xAudioCapturer.c [hunk 2]
```diff
@@ -0,0 +1,413 @@
+#include <stdlib.h>
```
Comment (codingspirit) [inline source/SV82x/SV82xAudioCapturer.c:1]: Files under *amazon-kinesis-video-streams-media-interface/source/* should be licensed under Apache 2.0. You may copy those license header from other files.

File: source/SV82x/SV82xAudioCapturer.c [hunk 3]
```diff
@@ -0,0 +1,413 @@
...
+		sv82xHandle->status = newStatus;
+		LOG("AudioCapturer new status[%d]", newStatus);
+	}
+
+	return 0;
+}
+
+AudioCapturerHandle audioCapturerCreate(void)
+{
+	printf("audioCapturerCreate ############################\n");
```
Comment (codingspirit) [inline source/SV82x/SV82xAudioCapturer.c:55]: Use `LOG` instead of `printf`

File: source/SV82x/SV82xAudioCapturer.c [hunk 4]
```diff
@@ -0,0 +1,413 @@
...
+		//stAiVqeAttr.stHpfCfg.enHpfFreq = AUDIO_HPF_FREQ_80;
+		stAiVqeAttr.u32OpenMask = g_audio_capture.audio_vqe.open_mask;
+		pAiVqeAttr = (EI_VOID *)&stAiVqeAttr;
+	}
+
+	s32Ret = SAMPLE_COMM_AUDIO_StartAi(AiDev, s32AiChnCnt, &stAioAttr,
+		g_audio_capture.enOutSampleRate, g_audio_capture.bAioReSample, pAiVqeAttr);
+	if (s32Ret != EI_SUCCESS) {
+		EI_TRACE_AUDIO(EI_DBG_ERR, "SAMPLE_COMM_AUDIO_StartAi failed, s32Ret=%d\n", s32Ret);
+		goto AIAENC_ERR2;
```
Comment (codingspirit) [inline source/SV82x/SV82xAudioCapturer.c:270]: Please avoid to use `goto`

File: source/SV82x/SV82xAudioCapturer.c [hunk 5]
```diff
@@ -0,0 +1,413 @@
...
+	for (i = 0; i < s32AencChnCnt; i++) {
+		AiChn = i;
+		s32Ret = SAMPLE_COMM_AUDIO_DestroyTrdAi(AiDev, AiChn);
+		if (s32Ret != EI_SUCCESS) {
+			EI_TRACE_AUDIO(EI_DBG_ERR, "SAMPLE_COMM_AUDIO_DestroyTrdAi failed, s32Ret=%d\n", s32Ret);
+		}
+
+		s32Ret = SAMPLE_COMM_AUDIO_DestroyTrdAencAdec(AdChn);
+		if (s32Ret != EI_SUCCESS) {
+			EI_TRACE_AUDIO(EI_DBG_ERR, "SAMPLE_COMM_AUDIO_DestroyTrdAencAdec failed, s32Ret=%d\n", s32Ret);
```
Comment (codingspirit) [inline source/SV82x/SV82xAudioCapturer.c:379]: `LOG`

File: source/SV82x/SV82xAudioCapturer.c [hunk 6]
```diff
@@ -0,0 +1,413 @@
...
+
+	s32Ret = SAMPLE_COMM_AUDIO_StopAenc(s32AencChnCnt);
+	if (s32Ret != EI_SUCCESS) {
+		printf("SAMPLE_COMM_AUDIO_StopAenc failed %x\n", s32Ret);
+		return s32Ret;
+	}
+
+	s32Ret = SAMPLE_COMM_AUDIO_StopAi(AiDev, s32AiChncnt, bAioReSample, bVqeEn);
+	if (s32Ret != EI_SUCCESS) {
+		printf("SAMPLE_COMM_AUDIO_StopAi failed %x\n", s32Ret);
```
Comment (codingspirit) [inline source/SV82x/SV82xAudioCapturer.c:391]: `LOG`

File: source/SV82x/SV82xAudioCapturer.c [hunk 7]
```diff
@@ -0,0 +1,413 @@
...
+	SV82XAudioCapturer *sv82xHandle =  (SV82XAudioCapturer *)malloc(sizeof(SV82XAudioCapturer));
+
+	if (!sv82xHandle) {
+		LOG("%s OOM", __func__);
+		return NULL;
+	}
+
+	memset(sv82xHandle, 0, sizeof(SV82XAudioCapturer));
+
+	// Now implementation supports raw PCM, G.711 ALAW and ULAW, MONO, 8k/16k, 16 bits
```
Comment (codingspirit) [inline source/SV82x/SV82xAudioCapturer.c:65]: I can see `AAC` related code below - Do you forget to add AAC into capability?

File: source/SV82x/SV82xAudioPlayer.c [hunk 8]
```diff
@@ -0,0 +1,353 @@
...
+	stAioAttr.u32ChnCnt	  = g_audio_player.paramDecode.u32ChnCnt;
+
+	FILE *pfo = NULL;
+
+	s32AoChnCnt = stAioAttr.u32ChnCnt;
+	s32Ret = SAMPLE_COMM_AUDIO_StartAo(AoDev, s32AoChnCnt, &stAioAttr,
+		g_audio_player.enInSampleRate, g_audio_player.bAioReSample);
+	if (s32Ret != EI_SUCCESS) {
+		EI_TRACE_AUDIO(EI_DBG_ERR, "SAMPLE_COMM_AUDIO_StartAo failed, ret=%d\n", s32Ret);
+		goto ADECAO_ERR2;
```
Comment (codingspirit) [inline source/SV82x/SV82xAudioPlayer.c:231]: remove `goto`

**Review by codingspirit — CHANGES_REQUESTED** [review]

**Review by codingspirit — APPROVED** [review]

**Pull request merged**